<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>975</width>
    <height>750</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QT Chat Studio</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="font">
       <font>
        <bold>true</bold>
        <stylestrategy>PreferAntialias</stylestrategy>
       </font>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="iconSize">
       <size>
        <width>16</width>
        <height>16</height>
       </size>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tab_main">
       <attribute name="title">
        <string>Main dialog</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="gb_chat">
          <property name="title">
           <string/>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="1" column="0">
            <widget class="QLineEdit" name="le_input">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>45</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>9</pointsize>
               <bold>false</bold>
               <stylestrategy>PreferAntialias</stylestrategy>
              </font>
             </property>
            </widget>
           </item>
           <item row="0" column="0" colspan="2">
            <widget class="QTextEdit" name="te_chat">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QPushButton" name="btn_send">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>0</height>
              </size>
             </property>
             <property name="font">
              <font>
               <pointsize>10</pointsize>
               <bold>true</bold>
               <stylestrategy>PreferAntialias</stylestrategy>
              </font>
             </property>
             <property name="cursor">
              <cursorShape>PointingHandCursor</cursorShape>
             </property>
             <property name="text">
              <string>Send</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>138</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QComboBox" name="cb_models"/>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_set">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QVBoxLayout" name="settingsMainLayout">
        <item>
         <widget class="QGroupBox" name="gb_ollama">
          <property name="title">
           <string>Ollama models</string>
          </property>
          <layout class="QVBoxLayout" name="modelsLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4">
             <item>
              <widget class="QPushButton" name="btn_update_models">
               <property name="text">
                <string>Update list</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_create_model">
               <property name="text">
                <string>Create model</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_delete_model">
               <property name="text">
                <string>Delete model</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QComboBox" name="cb_available_models"/>
           </item>
           <item>
            <widget class="QListWidget" name="lw_models">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>120</height>
              </size>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SelectionMode::MultiSelection</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gb_system_prompt">
          <property name="title">
           <string>System prompt</string>
          </property>
          <layout class="QVBoxLayout" name="promptLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QPushButton" name="btn_create_prompt">
               <property name="text">
                <string>Create new</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_save_prompt">
               <property name="text">
                <string>Save changes</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_delete_prompt">
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QComboBox" name="cb_saved_prompts">
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextEdit" name="te_system_prompt">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
             <property name="readOnly">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="gb_history">
          <property name="title">
           <string>Chat history</string>
          </property>
          <layout class="QVBoxLayout" name="historyLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6">
             <item>
              <widget class="QPushButton" name="btn_save_chat">
               <property name="text">
                <string>Save chat</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_open_chat">
               <property name="text">
                <string>Open saved</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_delete_chat">
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btn_export_chat">
               <property name="text">
                <string>Export</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QListWidget" name="lw_chat_history">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>120</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>975</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="action_new_chat"/>
    <addaction name="action_save_chat"/>
    <addaction name="action_open_file"/>
    <addaction name="separator"/>
    <addaction name="action_update"/>
   </widget>
   <addaction name="menufile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_new_chat">
   <property name="text">
    <string>New Chat</string>
   </property>
  </action>
  <action name="action_save_chat">
   <property name="text">
    <string>Save Chat</string>
   </property>
  </action>
  <action name="action_open_file">
   <property name="text">
    <string>Open File</string>
   </property>
  </action>
  <action name="action_update">
   <property name="text">
    <string>Update</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
